### ISSUE TITLE
error checking push permissions -- make sure you entered the correct tag name

### ISSUE THREAD

**Issue by eravindar12** [GoogleContainerTools/kaniko#2222]
I see below the issue and how to fix it.
 
 ```
$ export DOCKER_REPOSITORY=${DOCKER_REPOSITORY:-ecommerce/$CI_PROJECT_NAME${APP:+/}$APP}
$ echo "{\"auths\":{\"$DOCKER_REGISTRY\":{\"username\":\"$DFWP_HARBOR_USER\",\"password\":\"$DFWP_HARBOR_PASS\"}}}" > /kaniko/.docker/config.json
$ /kaniko/executor \ # collapsed multi-line command
error checking push permissions -- make sure you entered the correct tag name, and that you are authenticated correctly, and try again: checking push permission for "dfwpharb01.sial.com/ecommerce/aem-cms/aem-dispatcher:release-integration-3513e70d": POST https://dfwpharb01.sial.com/v2/ecommerce/aem-cms/aem-dispatcher/blobs/uploads/: UNAUTHORIZED: unauthorized to access repository: ecommerce/aem-cms/aem-dispatcher, action: push: unauthorized to access repository: ecommerce/aem-cms/aem-dispatcher, action: push

```

**Comment by iJebus** [comment]
I'm seeing similar with pushing to GCP artifact registry, looks like possible regression of https://github.com/GoogleContainerTools/kaniko/issues/1786? Per the comments on that issue, pushing works when using 1.6.0.

**Comment by vmamidala-conga** [comment]
ECR Push operation is failing with following error using kaniko, it used to work until yesterday.
Kaniko version - v1.6.0
I tried testing it out with v1.7.0, v1.9.0 (latest) but still facing same issue. Any one had same issue? and resolved? I really appreciate you help
`error checking push permissions -- make sure you entered the correct tag name, and that you are authenticated correctly, and try again: 
checking push permission for "https://ecr-private-repo/uploads": Post "https://ecr-private-repo/uploads": EOF`

**Comment by mofirouz** [comment]
Yep same issue here - works with v1.6.0, but any newer and it's borked.

**Comment by jessequinn** [comment]
v1.6.0 does not work for me .. all the way up to 1.12.1. It was working yesterday.

**Comment by kinchitagg** [comment]
I'm have the same issue. But surprisingly  it works for one of my jenkins pipeline but not for the other. Both are pushing on same ECR account just different repos.

**Comment by tirelibirefe** [comment]
> ECR Push operation is failing with following error using kaniko, it used to work until yesterday. Kaniko version - v1.6.0 I tried testing it out with v1.7.0, v1.9.0 (latest) but still facing same issue. Any one had same issue? and resolved? I really appreciate you help `error checking push permissions -- make sure you entered the correct tag name, and that you are authenticated correctly, and try again: checking push permission for "https://ecr-private-repo/uploads": Post "https://ecr-private-repo/uploads": EOF`

Hello @vmamidala-conga 
Were you able to fix the problem? I have the same problem; I tried v.1.16.0 and 1.17.0, nothing changed.

Thanks & Regards

**Comment by tirelibirefe** [comment]
> 1.5.1

1.12.1 didn't work at me.

**Comment by ALGHARRASS-AYOUB** [comment]
you must verify the encoded value of auth. if you use "username:password"  to authenticate you must encode it as :
echo -n "username:password" |  base64
and not : echo -n username:password |  base64.
because the double quotes makes the difference.
then you can put the content below to this location /kaniko/.docker/config.json
the content of the file:
```
{
       "auths": {
                "YOUR_CI_REGISTRY": {
                 "auth": ENCODED_BASE64_STRING"
                }
}
```

**Comment by tbernacchi** [comment]
same here

**Comment by hardikbansal** [comment]
Same problem here as well. Any fix for this? Facing the same problem with "pkg.dev" based google artifacts repositories as well. Not sure what is wrong here.
